Content.Goobstation.Shared/MartialArts/SharedMartialArtsSystem.PlasmaFist.cs

```diff
@@ -0,0 +1,178 @@
+        comp.Used = true;
+    }
+    #endregion
+
+    #region Combo Methods
+
+    private void OnTornadoSweep(Entity<CanPerformComboComponent> ent, ref TornadoSweepPerformedEvent args)
+    {
+        if (!_proto.TryIndex(ent.Comp.BeingPerformed, out var proto)
+            || !TryUseMartialArt(ent, proto, out var target, out var downed))
+            return;
+
+        var mapPos = _transform.GetMapCoordinates(ent).Position;
+        var hitPos = _transform.GetMapCoordinates(target).Position;
+        var dir = hitPos - mapPos;
+        dir *= 1f / dir.Length();
+
+        if (TryComp<PullableComponent>(target, out var pullable))
+            _pulling.TryStopPull(target, pullable, ent, true);
+
+        var xform = _entityManager.GetComponent<TransformComponent>(ent);
+        var mapId = xform.MapID;
+        var position = _transform.GetMapCoordinates(ent).Position;
+
+        var physicsQuery = _entityManager.GetEntityQuery<PhysicsComponent>();
+        var movedByPressureQuery = _entityManager.GetEntityQuery<MovedByPressureComponent>();
+
+        // Use the entity lookup system to find entities within the radius.
+        // Iterate over all entities with PhysicsComponent and TransformComponent
+        var entities = new List<EntityUid>();
+        var query = _entityManager.AllEntityQueryEnumerator<PhysicsComponent, TransformComponent>();
+        float repulseRadius = 10f;
+        while (query.MoveNext(out var uid, out var phys, out var trans))
+        {
+            //if (trans.MapID != mapId)
+            //    continue;
+
+            if (uid == ent.Owner)
+                break;
+
+            if ((position - _transform.GetMapCoordinates(uid).Position).LengthSquared() <= repulseRadius * repulseRadius)
+            {
+                entities.Add(uid);
+            }
+        }
+
+        foreach (var entity in entities)
+        {
+            if (!physicsQuery.TryGetComponent(entity, out var physics))
+                continue;
+
+            if (movedByPressureQuery.TryGetComponent(entity, out var movedPressure) && !movedPressure.Enabled) //Ignore magboots users
+                continue;
+
+            // Calculate the direction from the source to the entity.
+            var entityXform = _entityManager.GetComponent<TransformComponent>(entity);
+            var direction = _transform.GetMapCoordinates(entity).Position - position;
+            var distance = direction.Length();
+
+            // Normalize the direction.
+            if (distance > 0)
+            {
+                direction /= distance;
+            }
+            else
+            {
+                // If the entity is at the exact same position, apply a random direction.
+                direction = _random.NextVector2().Normalized();
+            }
+            // Apply the impulse.
+            _entityManager.System<SharedPhysicsSystem>().ApplyLinearImpulse(entity, direction * 1000, body: physics);
```

Comment:
Add SharedPhysicsSystem as entsys [Dependency] instead of doing this.
And also, you can just use GravityWellSystem - it does exactly the same thing as you have coded. But it is in server so use virtual methods or events.